src/arithmetic.jl

```diff
@@ -225,7 +227,7 @@ for T in (:HomogeneousPolynomial, :TaylorN)
         function *(a::Taylor1{T}, b::$T{Taylor1{S}}) where
                 {T<:NumberNotSeries, S<:NumberNotSeries}
 
-            @inbounds aux = a * b[1]
+            @inbounds aux = a * b.coeffs[1]
```

Comment:
`b[0]`?

Reply:
I think `b.coeffs[1]` here is correct, though it doesn't look so nice, since it covers both `b::HomogeneousPolynomial` (b[1]) and `b::TaylorN` (b[0]).